// nestjs-schedule/dist/scheduler.registry.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SchedulerRegistry = void 0;
const tslib_1 = require("tslib");
const common_1 = require("@nestjs/common");
const schedule_messages_1 = require("./schedule.messages");
let SchedulerRegistry = class SchedulerRegistry {
    constructor() {
        this.cronJobs = new Map();
        this.timeoutJobs = new Map();
        this.intervalJobs = new Map();
    }
    getCronJob(name) {
        return this.cronJobs.get(name);
    }
    getIntervalJob(name) {
        return this.intervalJobs.get(name);
    }
    getTimeoutJob(name) {
        return this.timeoutJobs.get(name);
    }
    addCronJob(name, cronJob) {
        const job = this.cronJobs.get(name);
        if (job) {
            throw new Error((0, schedule_messages_1.DUPLICATE_SCHEDULER)('Cron Job', name));
        }
        this.cronJobs.set(name, cronJob);
    }
    addIntervalJob(name, intervalJob) {
        const job = this.intervalJobs.get(name);
        if (job) {
            throw new Error((0, schedule_messages_1.DUPLICATE_SCHEDULER)('Interval', name));
        }
        this.intervalJobs.set(name, intervalJob);
    }
    addTimeoutJob(name, timeoutJob) {
        const job = this.timeoutJobs.get(name);
        if (job) {
            throw new Error((0, schedule_messages_1.DUPLICATE_SCHEDULER)('Timeout', name));
        }
        this.timeoutJobs.set(name, timeoutJob);
    }
    getCronJobs() {
        return this.cronJobs;
    }
    deleteCronJob(name) {
        var _a;
        const job = this.cronJobs.get(name);
        if (job) {
            (_a = job.job) === null || _a === void 0 ? void 0 : _a.cancel();
            this.cronJobs.delete(name);
        }
    }
    getIntervalJobs() {
        return this.intervalJobs;
    }
    deleteIntervalJob(name) {
        const job = this.intervalJobs.get(name);
        if (job) {
            clearInterval(job.ref);
            this.intervalJobs.delete(name);
        }
    }
    getTimeoutJobs() {
        return this.timeoutJobs;
    }
    deleteTimeoutJob(name) {
        const job = this.timeoutJobs.get(name);
        if (job) {
            clearTimeout(job.ref);
            this.timeoutJobs.delete(name);
        }
    }
};
SchedulerRegistry = tslib_1.__decorate([
    (0, common_1.Injectable)()
], SchedulerRegistry);
exports.SchedulerRegistry = SchedulerRegistry;
